esmvaltool/recipes/ref/recipe_ref_annual_cycle_region.yml

```
@@ -0,0 +1,149 @@
+# ESMValTool
+---
+documentation:
+  title: Benchmarking of a single model for a region through extract_shape.
+  description: >
+    Benchmarking: annual cycle.
+  authors:
+    - lauer_axel
+    - bock_lisa
+    - hassler_birgit
+    - lindenlaub_lukas
+    - schlund_manuel
+    - weigel_katja
+  maintainer:
+    - lauer_axel
+  references:
+    - lauer25gmd
+  projects:
+    - dlrmabak
+
+
+datasets:
+  # Dataset to be benchmarked
+  - {dataset: MIROC6, grid: gn, benchmark_dataset: true, plot_label: 'MIROC6'}
+
+preprocessors:
+  pp_abs:
+    regrid:
+      target_grid: 1x1
+      scheme: linear
+    extract_shape:
+      shapefile: ar6
+      crop: true
+      decomposed: true
+      ids:
+        Name:
+          - N.Europe
+    climate_statistics:
+      period: monthly
+    area_statistics:
+      operator: mean
+
+  pp_abs_200:
+    extract_levels:
+      levels: 20000
+      scheme: linear
+      coordinate: air_pressure
+    regrid:
+      target_grid: 1x1
+      scheme: linear
+    extract_shape:
+      shapefile: ar6
+      crop: true
+      decomposed: true
+      ids:
+        Name:
+          - N.Europe
+    climate_statistics:
+      period: monthly
+    area_statistics:
+      operator: mean
+
+
+diagnostics:
+  anncyc:
+    description: Plot global mean mean annual cycle for the given period including a reference dataset.
+    variables:
+      tas: &var_default_abs
+        start_year: 1980
+        end_year: 2009
+        preprocessor: pp_abs
+        project: CMIP6
+        mip: Amon
+        exp: historical
+        ensemble: r1i1p1f1
+        grid: gr
+        reference_dataset: HadCRUT5
+        additional_datasets:
+          - {dataset: HadCRUT5, project: OBS, type: ground, version: 5.0.1.0-analysis,
+             tier: 2, reference_for_metric: true'}
+      pr:
+        <<: *var_default_abs
+        reference_dataset: GPCP-SG
+        additional_datasets:
+          - {dataset: GPCP-SG, project: obs4MIPs, type: atmos, version: 2.3, tier: 1,
```

```suggestion
          - {dataset: GPCP-V2.3, project: obs4MIPs, tier: 1,
```
The name of the obs4MIPs dataset on ESGF is `GPCP-V2.3`. `type` is not a required facet for obs4MIPs data. The only available version on ESGF is `v20180519`, but the version does not need to be specified because ESMValCore will automatically use the latest available version.

---

You are right that `GPCP-SG` is used in the filename, but what matters is the value for the `source_id`, as that is what we call `dataset` in ESMValCore (see [here](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/api/esmvalcore.esgf.html#esmvalcore.esgf.facets.FACETS) for the mapping from 'our' facet names to ESGF names). [Using `source_id=GPCP-V2.3` in an ESGF search](https://esg-dn1.nsc.liu.se/esg-search/search?format=application%2Fsolr%2Bjson&limit=500&distrib=true&offset=0&type=File&project=obs4MIPs&source_id=GPCP-V2.3) returns a file:
```json
{
  "responseHeader":{
    "status":0,
    "QTime":5,
    "params":{
      "df":"text",
      "q.alt":"*:*",
      "indent":"true",
      "echoParams":"all",
      "fl":"*,score",
      "start":"0",
      "fq":["type:File",
        "project:\"obs4MIPs\"",
        "source_id:\"GPCP-V2.3\""],
      "sort":"id asc",
      "rows":"500",
      "q":"*:*",
      "shards":"localhost/solr/files,localhost:8985/solr/files,localhost:8986/solr/files,localhost:8987/solr/files,localhost:8988/solr/files,localhost:8989/solr/files,localhost:8990/solr/files",
      "tie":"0.01",
      "facet.limit":"-1",
      "qf":"text",
      "facet.method":"enum",
      "facet.mincount":"1",
      "wt":"json",
      "facet":"true",
      "facet.sort":"lex"}},
  "response":{"numFound":1,"start":0,"maxScore":1.0,"docs":[
      {
        "id":"obs4MIPs.NASA-GSFC.GPCP.atmos.mon.v20180519.pr_GPCP-SG_L3_v2.3_197901-201710.nc|dpesgf03.nccs.nasa.gov",
        "version":"1",
        "cf_standard_name":["precipitation_flux"],
        "checksum":["4dd4678b79ef139446c8406da5aae4fed210abb2f2160ef95f6988bf83e4525b"],
        "checksum_type":["SHA256"],
        "data_node":"dpesgf03.nccs.nasa.gov",
        "dataset_id":"obs4MIPs.NASA-GSFC.GPCP-V2.3.atmos.mon.v20180519|dpesgf03.nccs.nasa.gov",
        "dataset_id_template_":["%(project)s.%(institute)s.%(source_id)s.%(realm)s.%(time_frequency)s"],
        "index_node":"esgf-node.llnl.gov",
        "instance_id":"obs4MIPs.NASA-GSFC.GPCP.atmos.mon.v20180519.pr_GPCP-SG_L3_v2.3_197901-201710.nc",
        "institute":["NASA-GSFC"],
        "latest":true,
        "master_id":"obs4MIPs.NASA-GSFC.GPCP.atmos.mon.pr_GPCP-SG_L3_v2.3_197901-201710.nc",
        "model":["Obs-GPCP"],
        "product":["observations"],
        "project":["obs4MIPs"],
        "realm":["atmos"],
        "replica":false,
        "size":19348352,
        "source_id":["GPCP-V2.3"],
        "time_frequency":["mon"],
        "timestamp":"2018-02-17T17:17:54Z",
        "title":"pr_GPCP-SG_L3_v2.3_197901-201710.nc",
        "tracking_id":["4070c751-6c2d-440f-a4d7-5b325fb98990"],
        "type":"File",
        "url":["https://dpesgf03.nccs.nasa.gov/thredds/fileServer/obs4MIPs/observations/NASA-GSFC/Obs-GPCP/GPCP/V2.3/atmos/pr/pr_GPCP-SG_L3_v2.3_197901-201710.nc|application/netcdf|HTTPServer"],
        "variable":["pr"],
        "variable_long_name":["Precipitation"],
        "variable_units":["kg m-2 s-1"],
        "_version_":1662343940370595840,
        "retracted":false,
        "_timestamp":"2020-03-27T18:45:20.965Z",
        "score":1.0}]
  },
  "facet_counts":{
    "facet_queries":{},
    "facet_fields":{},
    "facet_ranges":{},
    "facet_intervals":{},
    "facet_heatmaps":{}}}
```
while a search [using `source_id=GPCP-SG` in an ESGF search](https://esg-dn1.nsc.liu.se/esg-search/search?format=application%2Fsolr%2Bjson&limit=500&distrib=true&offset=0&type=File&project=obs4MIPs&source_id=GPCP-SG) returns no files:
```json
{
  "responseHeader":{
    "status":0,
    "QTime":8,
    "params":{
      "df":"text",
      "q.alt":"*:*",
      "indent":"true",
      "echoParams":"all",
      "fl":"*,score",
      "start":"0",
      "fq":["type:File",
        "project:\"obs4MIPs\"",
        "source_id:\"GPCP-SG\""],
      "sort":"id asc",
      "rows":"500",
      "q":"*:*",
      "shards":"localhost/solr/files,localhost:8985/solr/files,localhost:8986/solr/files,localhost:8987/solr/files,localhost:8988/solr/files,localhost:8989/solr/files,localhost:8990/solr/files",
      "tie":"0.01",
      "facet.limit":"-1",
      "qf":"text",
      "facet.method":"enum",
      "facet.mincount":"1",
      "wt":"json",
      "facet":"true",
      "facet.sort":"lex"}},
  "response":{"numFound":0,"start":0,"maxScore":0.0,"docs":[]
  },
  "facet_counts":{
    "facet_queries":{},
    "facet_fields":{},
    "facet_ranges":{},
    "facet_intervals":{},
    "facet_heatmaps":{}}}
```

---

But when I write "dataset: GPCP-V2.3" in the recipe it does not run (maybe it would, if I'd allow download?) but it does with dataset: GPCP-SG (because on Levante, these data are available)?

---

Thanks for the answer @bouweandela ! I tried again with `search_esgf: when_missing` in config, then it works with "GPCP-V2.3". But it does not work any more, when I set it to ` search_esgf: never ` to run the next recipe, which needs the same data (or even to rerun the same recipe again). Then it still does not find the data? The reason seem to be, that it does not create a folder. It puts the data under:
(esmvaltool) [b380216@levante1 ESMValTool]$ ls -lhrt /work/bd1083/b380216/extraobsv2/Tier1/
total 19M
drwxr-sr-x 2 b380216 bd0854 20K Jun 15 12:28 ERA-5
-rwx------ 1 b380216 bd0854 19M Jun 25 15:39 pr_GPCP-SG_L3_v2.3_197901-201710.nc

When I move the data by hand into a sub-folder called "GPCP-V2.3" it works, so I change it in these recipes now to GPCP-V2.3 now.
But I wonder if we need to change something for the downloader in cases where file name and the source_id disagree
and/or if we then should rename?
/work/bd0854/DATA/ESMValTool2/OBS/Tier1/GPCP-SG
to
/work/bd0854/DATA/ESMValTool2/OBS/Tier1/GPCP-V2.3
? 
But this would cause other recipes, which currently use "GPCP-SG" not to work anymore on Levante?
These recipes are:
- esmvaltool/recipes/recipe_model_benchmarking_timeseries_pr_e.yml
- esmvaltool/recipes/examples/recipe_check_obs.yml
- esmvaltool/recipes/ipccwg1ar5ch9/recipe_flato13ipcc_figure_96.yml
- esmvaltool/recipes/ipccwg1ar6ch3/recipe_ipccwg1ar6ch3_atmosphere.yml
- esmvaltool/recipes/model_evaluation/recipe_model_benchmarking_boxplots.yml
- esmvaltool/recipes/model_evaluation/recipe_model_benchmarking_maps.yml
- esmvaltool/recipes/model_evaluation/recipe_model_evaluation_basics.yml
- esmvaltool/recipes/model_evaluation/recipe_model_evaluation_clouds_clim.yml
- esmvaltool/recipes/model_evaluation/recipe_model_evaluation_precip_zonal.yml
-esmvaltool/recipes/ref/recipe_ref_scatterplot.yml

There is also still the issue, that I cannot find pr and hus as cmorized ERA5 data?

---

> But it does not work any more, when I set it to `search_esgf: never` to run the next recipe, which needs the same data (or even to rerun the same recipe again). 

You need to set the rootpath and DRS so it matches the directory tree in your config-user.yml file. For example, mine looks like this for the obs4MIPs project on Levante:
```yaml
rootpath:
  obs4MIPs:
    /work/bd0854/DATA/ESMValTool2/OBS: default
    /work/bd0854/DATA/ESMValTool2/download: ESGF
```
> I wonder if we need to change something for the downloader

The downloader uses DRS `ESGF`, for obs4MIPs that is https://github.com/ESMValGroup/ESMValCore/blob/4d8fcbfb331e0745db58321cc41fe7de88c82393/esmvalcore/config-developer.yml#L110-L119
to store files. As you can see, the only requirement on the filename is that it starts with the `short_name` for that DRS setting, so what comes after it should't matter.

---

There is no issue with the file name, but with the name of the directory, which is usually the data set name:
It downloads the data to:
Tier{tier}/
but expects: 'Tier{tier}/{dataset}' for finding the data, if they are not downloaded in the same run.
I'd expect that most users don't change config_developer (because it is part of ESMValCore) so if I use `dataset: GPCP-V2.3` it expects this as directory name and won't find the data if they are directly under Tier{tier}/ or in `/work/bd0854/DATA/ESMValTool2/OBS/Tier1/GPCP-SG` ?